hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/txns/crypto/validators/CryptoCreateChecks.java

```diff
@@ -179,53 +179,51 @@ private ResponseCodeEnum validateKeyAliasAndEvmAddressCombinations(
         if (onlyKeyProvided(op)) {
             return validateKey(op);
-        } else if (onlyEvmAddressProvided(op)) {
-            return validateOnlyEvmAddressProvidedCase(op);
         } else if (onlyAliasProvided(op)) {
             return validateOnlyAliasProvidedCase(op);
         } else if (keyAndAliasProvided(op)) {
             return validateKeyAndAliasProvidedCase(op);
-        } else if (keyAndEvmAddressProvided(op)) {
-            return validateKeyAndEvmAddressProvidedCase(op);
-        } else if (aliasAndEvmAddressProvided(op)) {
-            return validateAliasAndEvmAddressProvidedCase(op);
-        } else if (keyAndAliasAndEvmAddressProvided(op)) {
-            return validateKeyAndAliasAndEVMAddressProvidedCase(op);
         } else {
-            // This is the case when no key, no alias and no evm address are provided
+            // This is the case when no key and no alias are provided
             return KEY_REQUIRED;
         }
     }
 
     public static boolean onlyKeyProvided(CryptoCreateTransactionBody op) {
-        return op.hasKey() && op.getAlias().isEmpty() && op.getEvmAddress().isEmpty();
+        return op.hasKey() && op.getAlias().isEmpty();
     }
 
     public static boolean keyAndAliasProvided(CryptoCreateTransactionBody op) {
-        return op.hasKey() && !op.getAlias().isEmpty() && op.getEvmAddress().isEmpty();
+        return op.hasKey() && !op.getAlias().isEmpty();
     }
 
-    public static boolean keyAndAliasAndEvmAddressProvided(CryptoCreateTransactionBody op) {
-        return op.hasKey() && !op.getAlias().isEmpty() && !op.getEvmAddress().isEmpty();
+    public static boolean onlyAliasProvided(CryptoCreateTransactionBody op) {
+        return !op.hasKey() && !op.getAlias().isEmpty();
     }
 
-    public static boolean keyAndEvmAddressProvided(CryptoCreateTransactionBody op) {
-        return op.hasKey() && op.getAlias().isEmpty() && !op.getEvmAddress().isEmpty();
-    }
+    private ResponseCodeEnum validateKeyAndAliasProvidedCase(final CryptoCreateTransactionBody op) {
+        if (op.getAlias().size() == EVM_ADDRESS_SIZE) {
+            if (!dynamicProperties.isCryptoCreateWithAliasAndEvmAddressEnabled()) {
+                return NOT_SUPPORTED;
+            }
 
-    public static boolean onlyAliasProvided(CryptoCreateTransactionBody op) {
-        return !op.hasKey() && !op.getAlias().isEmpty() && op.getEvmAddress().isEmpty();
-    }
+            if (HederaEvmContractAliases.isMirror(op.getAlias().toByteArray())) {
+                return INVALID_ALIAS_KEY;
+            }
 
-    public static boolean aliasAndEvmAddressProvided(CryptoCreateTransactionBody op) {
-        return !op.hasKey() && !op.getAlias().isEmpty() && !op.getEvmAddress().isEmpty();
-    }
+            final var keyValidity = validateKey(op);
+            if (keyValidity != OK) {
+                return keyValidity;
+            }
 
-    public static boolean onlyEvmAddressProvided(CryptoCreateTransactionBody op) {
```

Comment:
(Anyone else noticing in this review that Github's diff it particularly confused here?)